unikernel.ml

```diff
@@ -530,7 +586,187 @@ struct
             ~icon:"/images/robur.png" ())
          `OK)
 
-  let users store reqd user =
+  let account_page store reqd (user : User_model.user) =
+    match Middleware.get_cookie_from_request reqd with
+    | Ok cookie_value -> (
+        match User_model.user_auth_cookie_from_user cookie_value user with
+        | Some cookie -> (
+            let now = Ptime.v (P.now_d_ps ()) in
+            generate_csrf_token store user now reqd >>= function
+            | Ok csrf ->
+                Lwt.return
+                  (reply reqd ~content_type:"text/html"
+                     (Dashboard.dashboard_layout user
+                        ~page_title:"Account | Mollymawk"
+                        ~content:
+                          (User_account.user_account_layout ~csrf user cookie
+                             now)
+                        ~icon:"/images/robur.png" ())
+                     ~header_list:[ ("X-MOLLY-CSRF", csrf) ]
+                     `OK)
+            | Error err ->
+                Lwt.return
+                  (reply reqd ~content_type:"text/html"
+                     (Dashboard.dashboard_layout user
+                        ~page_title:"500 | Mollymawk"
+                        ~content:(Error_page.error_layout err)
+                        ~icon:"/images/robur.png" ())
+                     `Internal_server_error))
+        | None ->
+            let error =
+              {
+                Utils.Status.code = 401;
+                title = "Unauthenticated";
+                success = false;
+                data = "Auth cookie not found";
+              }
+            in
+            Lwt.return
+              (reply reqd ~content_type:"text/html"
+                 (Dashboard.dashboard_layout user ~page_title:"401 | Mollymawk"
+                    ~content:(Error_page.error_layout error)
+                    ~icon:"/images/robur.png" ())
+                 `Unauthorized))
+    | Error (`Msg err) ->
+        let error =
+          {
+            Utils.Status.code = 401;
+            title = "Unauthenticated";
+            success = false;
+            data = err;
+          }
+        in
+        Lwt.return
+          (reply reqd ~content_type:"text/html"
+             (Dashboard.dashboard_layout user ~page_title:"401 | Mollymawk"
+                ~content:(Error_page.error_layout error)
+                ~icon:"/images/robur.png" ())
+             `Unauthorized)
```

Comment:
hmm, isn't the authentication already protected / guaranteed by the main dispatcher?

Reply:
(partially) done in 6228aec